app/scripts/lib/nonce-tracker.js

```diff
@@ -28,18 +29,23 @@ class NonceTracker {
     // calculate next nonce
     // we need to make sure our base count
     // and pending count are from the same block
+    const localNonceHex = this._getHighestLocalNonce(address)
+    let localNonce = parseInt(localNonceHex, 16)
+    // throw out localNonce if not a number
+    if (!Number.isInteger(localNonce)) localNonce = 0
     const currentBlock = await this._getCurrentBlock()
     const pendingTransactions = this.getPendingTransactions(address)
     const pendingCount = pendingTransactions.length
     assert(Number.isInteger(pendingCount), `nonce-tracker - pendingCount is not an integer - got: (${typeof pendingCount}) "${pendingCount}"`)
     const baseCountHex = await this._getTxCount(address, currentBlock)
     const baseCount = parseInt(baseCountHex, 16)
     assert(Number.isInteger(baseCount), `nonce-tracker - baseCount is not an integer - got: (${typeof baseCount}) "${baseCount}"`)
-    const nextNonce = baseCount + pendingCount
+    if (localNonce) ++localNonce
```

Comment:
this will give us the wrong number if there is 1 confirmed tx and its nonce is 0